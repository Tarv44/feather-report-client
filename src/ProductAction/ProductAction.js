import React, {Component} from 'react';
import styles from './ProductAction.module.css';
import ProductContext from '../productContext';

export default class ProductAction extends Component {
    static defaultProps = {
        handleForm: () => {console.log('Error: Set handleForm.')}
    }

    static contextType = ProductContext

    state = {
        selected: false
    }

    onSelect = () => {
        if (this.state.selected) {
            this.setState({ selected: false })
            this.context.handleSelected('remove', this.props.product)
        } else {
            if (this.context.selected.length < 4) {
                this.setState({ selected: true })
                this.context.handleSelected('add', this.props.product)
            }
        }
    }

    setDisplay() {
        const type = this.props.type
        if (type === 'edit') {
            return (
                <button
                    className={`${styles.button } ${styles.edit}`}
                    onClick={e => this.props.handleForm(e)}
                >
                    Edit Product
                </button>
            )
        }

        if (type === 'select') {
            if (this.state.selected) {
                return <button 
                    onClick={e => this.onSelect()}
                    className={`${styles.button } ${styles.selected}`}
                >
                    Selected
                </button>
            }
            return (
                <button
                    className={`${styles.button } ${styles.select}`}
                    onClick={e => this.onSelect()}
                >
                    Select Product
                </button>
            )
        }

        if (type === 'compare') {
            return (
                <button
                    className={`${styles.button } ${styles.compare}`}
                >
                    Buy Online
                </button>
            )
        }
    }

    render() {
        const action = this.setDisplay()

        return action
    }
}