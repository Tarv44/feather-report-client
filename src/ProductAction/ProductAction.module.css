.button {
    font-size: 16px;
    background-color: white;
    border: solid black 3px;
    /* box-shadow: 3px 3px 1px #ccc; */
    margin-bottom: 20px;
}