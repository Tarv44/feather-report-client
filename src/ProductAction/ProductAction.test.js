import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import ProductAction from './ProductAction'

describe('----- ProductAction -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(<ProductAction type='edit'/>, div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(<ProductAction type='edit'/>)
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})