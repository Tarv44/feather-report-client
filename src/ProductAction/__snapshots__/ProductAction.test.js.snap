// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- ProductAction ----- renders the UI as expected 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Edit Product
</button>
`;
